File: airesources/Java/hlt/Move.java
package hlt;

public class Move {

    public enum MoveType { Noop, Thrust, Dock, Undock }

    private final MoveType type;
    private final Ship ship;

    public Move(final MoveType type, final Ship ship) {
        this.type = type;
        this.ship = ship;
    }

    public MoveType getType() {
        return type;
    }

    public Ship getShip() {
        return ship;
    }
}

File: airesources/Scala/hlt/Move.scala
package hlt

object Move {

  def values: List[MoveType] = List(Noop, Thrust, Dock, Undock)

  sealed trait MoveType

  case object Noop extends MoveType

  case object Thrust extends MoveType

  case object Dock extends MoveType

  case object Undock extends MoveType

}

class Move(val moveType: Move.MoveType, val ship: Ship)
